### ISSUE TITLE
Bifrost failed to create Maya attribute for type BifrostUsd::Stage

### ISSUE THREAD

**Issue by dgovil** [Autodesk/bifrost-usd#5]
I'm hitting an issue with Bifrost USD when compiled from source against Maya 2023.2 with USD 22.11 and Maya USD 0.21 on macOS where I cannot connect the output of the `Add to Stage` node to the `output` node.

When I try and do it, I get an error: `Bifrost failed to create Maya attribute for type BifrostUsd::Stage`.
I can't seem to find that string anywhere in the source code, so I was hoping someone could help point me as to where to look for what might be causing the issue? 

<img width="1348" alt="image" src="https://user-images.githubusercontent.com/4443090/210454507-09fa9620-5f34-45e7-ab8b-582ac20041b2.png">

However if I try loading Maya 2023.2 with the default bifrost + usd version that it ships with I am able to do so like in the graph below.
<img width="1454" alt="image" src="https://user-images.githubusercontent.com/4443090/210453873-6a0342f8-065e-42dd-9967-c376ba90b64f.png">

**Comment by frenchdog** [comment]
Hi Dhruv,
This generally happen when Bifrost USD and Maya USD are not built using
same USD libs. Is it your case? Even if they are the same version,
different compilation flags, namespaces, etc. could give such error.

**Comment by dgovil** [comment]
Hmm in our case they're built against the same build of USD (and we control it via Rez so there shouldn't be any deviance)
Looking at the linker paths, it all looks good but of course it's possible I'm missing it in one spot.

We also build with the default namespace, so I'd expect to see the usual linker warnings when symbol conflicts occur (but also entirely possible that's getting eaten somewhere)

Could you point me to the callback that happens when it tries to connect `out_stage` to then create the UFE stage? I can start using that to dig into what might be happening

**Comment by frenchdog** [comment]
The stage created by Bifrost USD is passed to Maya USD using its cache id. It is done in this function https://github.com/Autodesk/bifrost-usd/blob/760ada6833134cb460424e0d63e50c7b85691312/maya_plugin/src/usdmodule/usd_translator.cpp#L66
Are you passing MAYA_RUNTIME_LOCATION in the cmake command? Is you install of Bifrost USD includes the usd_maya_translation.json file?

**Comment by frenchdog** [comment]
And maybe in your case it is failing before UsdTranslation::convertValueToHost.
You can also take a look here: https://github.com/Autodesk/bifrost-usd/blob/760ada6833134cb460424e0d63e50c7b85691312/maya_plugin/src/usdmodule/usd_translator.cpp#L106

**Comment by dgovil** [comment]
Hmm I'm not passing in `MAYA_RUNTIME_LOCATION`. I only pass in the following since we have our own USD build in a different Rez managed location.

```
CMAKE_BUILD_TYPE
BIFROST_LOCATION
BIFUSD_PACKAGE_NAME
USD_LOCATION
CMAKE_INSTALL_PREFIX
PYTHON_VERSION_MAJOR
PYTHON_VERSION_MINOR
PYTHON_LIBRARY
PYTHON_EXECUTABLE
PYTHON_INCLUDE_DIR
```

I do not see a `usd_maya_translation.json` under the installed location. Only the `usd_pack_config.json` and the various node definition json files. But that gives me a good thing to start with debugging

**Comment by frenchdog** [comment]
Ok so this is why it does not work!
Bifrost USD nodes can be used without Maya (like if you run the command line tool bifcmd) but when in Maya you need more than just the USD nodes (provided by usd_pack_config.json). 
The usd_maya_translation.json need to be added to your config.
It is mentioned in the README here [Loading the USD Pack in Bifrost Extension for Maya](https://github.com/Autodesk/bifrost-usd/tree/760ada6833134cb460424e0d63e50c7b85691312#loading-the-usd-pack-in-bifrost-extension-for-maya)
In short, this is the plugin_config.json file that need to be added to BIFROST_LIB_CONFIG_FILES. This json file includes usd_pack_config.json and usd_maya_translation.json.

**Comment by dgovil** [comment]
Ah my bad. I got to `Loading the USD Pack in Bifrost Extension for Maya`, got it loading the nodes in Maya and then missed the next step. 

I'll give that a go and close out the issue once it's working.

**Comment by dgovil** [comment]
Ah excellent. That worked. I'll close this out, though might I suggest changing the description for `MAYA_RUNTIME_LOCATION` to say it's required for use in Maya?

In hindsight it's obvious, but when I was setting this up, I read that it was `optional` and left it out.

Thanks very much for the troubleshooting!

**Comment by frenchdog** [comment]
Very happy to hear that you figure it out!
Indeed we should definitely improve the README to avoid such confusion.
Thanks for your feedback!
